Follow WG Naming Recommendations on Master -> Control Plane

gjtempleton opened the pull request:
Updating documentation references to `master` to control plane/API Server as appropriate as per https://github.com/kubernetes/community/blob/master/wg-naming/recommendations/001-master-control-plane.md

A number of example files are renamed to match this convention. 

I've left explicit references to the current `master` taint where discussed as part of allowing the CA to run on a control plane node as this is deprecated, but not yet removed as per: https://github.com/kubernetes/kubernetes/pull/95382

Fixes #3796

mwielgus reviewed (changes requested):
As many people may not be familiar what "control plane" node is (we have been using `master` for ages), please add something like `(previously known as "master node")` where applicable.

gjtempleton commented:
@mwielgus It feels a bit messy to me, but I've added clarifying text anywhere I moved from `master` to `control-plane` along with some hidden `TODO:` stanzas to remind us to come back and remove these once k8s fully removes the terminology.

mwielgus reviewed (approved):
/lgtm
/approve
